### PR TITLE
WLTS-27 - Exception handler to present consistent error responses

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/WLTS-27

### Change description ###

Return formatted errors, in same format as CPO, see below

```
{
    "exception" : "org.springframework.web.bind.MethodArgumentNotValidException",
    "timestamp" : "2022-05-31T16:08:40.986759",
    "status" : 400,
    "error" : "Bad Request",
    "message" : "Input not valid",
    "path" : "/case-payment-orders",
    "details" : [ "Case ID has to be a valid 16-digit Luhn number." ]
}

```
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        setupMockServiceToThrowException(expectedException);
+        ResultActions result = this.mockMvc.perform(get(DICTIONARY_URL)
+                                                        .contentType(MediaType.APPLICATION_JSON));
+
+        // THEN
+        assertHttpErrorResponse(result, HttpStatus.FORBIDDEN, expectedException.getMessage());
+
+    }
+
+    @DisplayName("should return correct response when CasePaymentOrdersFilterException is thrown")
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandlerTest.java:122]: Can we reword

File: src/main/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        return ResponseEntity
+            .status(HttpStatus.BAD_REQUEST)
+            .body(error);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleHttpMessageNotReadable(
+        HttpMessageNotReadableException exception, HttpHeaders headers, HttpStatus status, WebRequest request) {
+
+        log.debug("MethodArgumentNotValidException:{}", exception.getLocalizedMessage());
```
Comment (mattnayler) [inline src/main/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandler.java:70]: Can we reword exception name in debug message to match trapped exception: `HttpMessageNotReadableException`

**Review by mattnayler — CHANGES_REQUESTED** [review]
Approved:

File: src/test/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandlerTest.java (hunk 1 above)
Comment (sullivangareth) [reply src/test/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandlerTest.java]: Fixed in [f64e292](https://github.com/hmcts/ts-translation-service/pull/51/commits/f64e292233d45defe1b478426cae0b933d1bf0e6)

File: src/main/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandler.java (hunk 2 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/translate/errorhandling/RestExceptionHandler.java]: Fixed in [6bbc198](https://github.com/hmcts/ts-translation-service/pull/51/commits/6bbc198413c4d31eff009bae0d5dc04f9ecf2407)

**Review by mattnayler — APPROVED** [review]
LGTM 👍

**Review by mattnayler — APPROVED** [review]
Re-approved after minor updates made from checkpoint discussions.

**Pull request merged**